### ISSUE TITLE
java client. LocalStorage does not pass parameters to some javascripts.

### ISSUE THREAD

**Issue by kotrachev** [SeleniumHQ/selenium#5663]
## Meta -
OS:  Windows 10
Selenium Version: java client 3.11.0 (https://www.seleniumhq.org/download/  selenium client Java 3.11.0 2018-03-11)

Browser:  no matter

Browser Version:  no matter

## Expected Behavior -
WebDriver java client must send to
URI: /session/96d0276c-d355-4a78-8970-de243db0714b/execute/sync
Body: {"args":["privet-key"],"script":"localStorage.removeItem('arguments[0]')"}

## Actual Behavior -
WebDriver java client actually sends
URI: /session/96d0276c-d355-4a78-8970-de243db0714b/execute/sync
Body: {"args":[],"script":"localStorage.removeItem('arguments[0]')"}

note empty args array in body JSON.

## Steps to reproduce -
java WebDriver-client code:
// create session to any W3C WebDriver-server
final URL url = new URL(addr);//"http://localhost:8085");
final DesiredCapabilities capabilities = new DesiredCapabilities();
capabilities.setBrowserName(...);
driver = new RemoteWebDriver(url, capabilities); 

final RemoteExecuteMethod executeMethod = new RemoteExecuteMethod(driver);
final RemoteWebStorage webStorage = new RemoteWebStorage(executeMethod);
final LocalStorage storage = webStorage.getLocalStorage();
storage.removeItem(key); // fails

## Thoughts -
W3CHttpCommandCodec.java passes javascript text to toScript function, but for some reason does not pass arguments. It seems it should pass arguments as well where appropriate. Like this:
// W3CHttpCommandCodec.java line 249
case SET_LOCAL_STORAGE_ITEM:
return toScript("localStorage.setItem('arguments[0]', 'arguments[1]')", parameters.get("key"), parameters.get("value"));

case REMOVE_LOCAL_STORAGE_ITEM:
return toScript("localStorage.removeItem('arguments[0]')", parameters.get("key"));

case GET_LOCAL_STORAGE_ITEM:
return toScript("return localStorage.getItem('arguments[0]')", parameters.get("key"));

// The same about _SESSION_STORAGE_.

Found it in backwoods of LOL (logic obscure layer).

**Comment by kotrachev** [comment]
Simple workaround - https://gist.github.com/roydekleijn/5073579

public String getKeyFromLocalStorage(int key) { 
    return (String) js.executeScript(String.format("return window.localStorage.key('%s');", key)); 
}

**Comment by barancev** [comment]
For issues please provide a [concise reproducible test case](http://sscce.org/) and describe what results you are seeing and what results you expect.

See [CONTRIBUTING.md](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#issue-contributions)

**Comment by barancev** [comment]
I'm trying to reproduce the issue. Here is the code:
```
RemoteWebDriver driver = new RemoteWebDriver(new ChromeOptions());
driver.get("https://seleniumhq.org/");
final RemoteExecuteMethod executeMethod = new RemoteExecuteMethod(driver);
final RemoteWebStorage webStorage = new RemoteWebStorage(executeMethod);
final LocalStorage storage = webStorage.getLocalStorage();
storage.setItem("aaa", "xxx");
System.out.println("before removeItem:" + storage.getItem("aaa"));
storage.removeItem("aaa");
System.out.println("after removeItem:" + storage.getItem("aaa"));
```
It prints out:
```
before removeItem:xxx
after removeItem:null
```
and the server debug log is:
```
21:44:48.188 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 742a45c469f7804b216357a5d4fcb714 (chrome): Executing POST on /session/742a45c469f7804b216357a5d4fcb714/local_storage (handler: ServicedSession)
21:44:48.190 DEBUG [Passthrough.handle] - To upstream: {"key":"aaa","value":"xxx"}
21:44:48.190 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@64884b6a10 pairs: {POST /session/742a45c469f7804b216357a5d4fcb714/local_storage HTTP/1.1: null}{User-Agent: okhttp/3.9.1}{Accept-Encoding: gzip}{Content-Type: application/json; charset=utf-8}{Connection: close}{Cache-Control: no-cache}{Pragma: no-cache}{Host: localhost:17122}{Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2}{Content-Length: 27}
21:44:48.227 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@18a6d32b4 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 72}{Content-Type: application/json; charset=utf-8}{Connection: close}
21:44:48.228 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"742a45c469f7804b216357a5d4fcb714","status":0,"value":null}
21:44:48.232 DEBUG [WebDriverServlet.handle] - Found handler: 742a45c469f7804b216357a5d4fcb714 (org.openqa.selenium.chrome.ChromeDriverService)
21:44:48.233 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 742a45c469f7804b216357a5d4fcb714 (chrome): Executing GET on /session/742a45c469f7804b216357a5d4fcb714/local_storage/key/aaa (handler: ServicedSession)
21:44:48.233 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@682174d88 pairs: {GET /session/742a45c469f7804b216357a5d4fcb714/local_storage/key/aaa HTTP/1.1: null}{User-Agent: okhttp/3.9.1}{Accept-Encoding: gzip}{Cache-Control: no-cache}{Connection: close}{Pragma: no-cache}{Host: localhost:17122}{Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2}
21:44:48.245 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@4e0861a4 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 73}{Content-Type: application/json; charset=utf-8}{Connection: close}
21:44:48.246 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"742a45c469f7804b216357a5d4fcb714","status":0,"value":"xxx"}
21:44:48.251 DEBUG [WebDriverServlet.handle] - Found handler: 742a45c469f7804b216357a5d4fcb714 (org.openqa.selenium.chrome.ChromeDriverService)
21:44:48.251 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 742a45c469f7804b216357a5d4fcb714 (chrome): Executing DELETE on /session/742a45c469f7804b216357a5d4fcb714/local_storage/key/aaa (handler: ServicedSession)
21:44:48.252 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@22955ef78 pairs: {DELETE /session/742a45c469f7804b216357a5d4fcb714/local_storage/key/aaa HTTP/1.1: null}{User-Agent: okhttp/3.9.1}{Accept-Encoding: gzip}{Connection: close}{Cache-Control: no-cache}{Pragma: no-cache}{Host: localhost:17122}{Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2}
21:44:48.256 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@c84ee054 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 72}{Content-Type: application/json; charset=utf-8}{Connection: close}
21:44:48.256 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"742a45c469f7804b216357a5d4fcb714","status":0,"value":null}
21:44:48.258 DEBUG [WebDriverServlet.handle] - Found handler: 742a45c469f7804b216357a5d4fcb714 (org.openqa.selenium.chrome.ChromeDriverService)
21:44:48.259 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 742a45c469f7804b216357a5d4fcb714 (chrome): Executing GET on /session/742a45c469f7804b216357a5d4fcb714/local_storage/key/aaa (handler: ServicedSession)
21:44:48.259 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@6f60dcf98 pairs: {GET /session/742a45c469f7804b216357a5d4fcb714/local_storage/key/aaa HTTP/1.1: null}{User-Agent: okhttp/3.9.1}{Accept-Encoding: gzip}{Cache-Control: no-cache}{Connection: close}{Pragma: no-cache}{Host: localhost:17122}{Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2}
21:44:48.264 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@6f3057594 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 72}{Content-Type: application/json; charset=utf-8}{Connection: close}
21:44:48.265 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"742a45c469f7804b216357a5d4fcb714","status":0,"value":null}
```
That does not match your description.

**Comment by barancev** [comment]
Fixed by f45fe9bb320f7e8a0dd44eeae76377b7bb895104
To be released in version 3.12
